[2025-01-31T19:14:11.417+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-31T19:14:11.450+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pyspark_example.run_pyspark_job manual__2025-01-31T19:14:10.811682+00:00 [queued]>
[2025-01-31T19:14:11.467+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pyspark_example.run_pyspark_job manual__2025-01-31T19:14:10.811682+00:00 [queued]>
[2025-01-31T19:14:11.468+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2025-01-31T19:14:11.496+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): run_pyspark_job> on 2025-01-31 19:14:10.811682+00:00
[2025-01-31T19:14:11.507+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=256) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-01-31T19:14:11.510+0000] {standard_task_runner.py:72} INFO - Started process 258 to run task
[2025-01-31T19:14:11.508+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'pyspark_example', 'run_pyspark_job', 'manual__2025-01-31T19:14:10.811682+00:00', '--job-id', '10', '--raw', '--subdir', 'DAGS_FOLDER/spark_dag.py', '--cfg-path', '/tmp/tmp_hk5ypgy']
[2025-01-31T19:14:11.516+0000] {standard_task_runner.py:105} INFO - Job 10: Subtask run_pyspark_job
[2025-01-31T19:14:11.539+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:209 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-01-31T19:14:11.608+0000] {task_command.py:467} INFO - Running <TaskInstance: pyspark_example.run_pyspark_job manual__2025-01-31T19:14:10.811682+00:00 [running]> on host bbdff58e2265
[2025-01-31T19:14:11.730+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='pyspark_example' AIRFLOW_CTX_TASK_ID='run_pyspark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-01-31T19:14:10.811682+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-31T19:14:10.811682+00:00'
[2025-01-31T19:14:11.731+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-31T19:14:11.790+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2025-01-31T19:14:11.792+0000] {spark_submit.py:401} INFO - Spark-Submit cmd: spark-submit --master yarn --executor-cores 1 --executor-memory 1g --driver-memory 1g --name pyspark_***_job --verbose --queue root.default --deploy-mode client /opt/***/dags/scripts/pyspark_job.py
[2025-01-31T19:14:11.860+0000] {spark_submit.py:571} INFO - JAVA_HOME is not set
[2025-01-31T19:14:11.872+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 502, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --executor-cores 1 --executor-memory 1g --driver-memory 1g --name pyspark_***_job --verbose --queue root.default --deploy-mode client /opt/***/dags/scripts/pyspark_job.py. Error code is: 1.
[2025-01-31T19:14:11.883+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=pyspark_example, task_id=run_pyspark_job, run_id=manual__2025-01-31T19:14:10.811682+00:00, execution_date=20250131T191410, start_date=20250131T191411, end_date=20250131T191411
[2025-01-31T19:14:11.901+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-31T19:14:11.902+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 10 for task run_pyspark_job (Cannot execute: spark-submit --master yarn --executor-cores 1 --executor-memory 1g --driver-memory 1g --name pyspark_***_job --verbose --queue root.default --deploy-mode client /opt/***/dags/scripts/pyspark_job.py. Error code is: 1.; 258)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 502, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --executor-cores 1 --executor-memory 1g --driver-memory 1g --name pyspark_***_job --verbose --queue root.default --deploy-mode client /opt/***/dags/scripts/pyspark_job.py. Error code is: 1.
[2025-01-31T19:14:11.946+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-31T19:14:11.974+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-31T19:14:11.978+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
